### PR TITLE
Add Front Door redirect template

### PR DESCRIPTION
**Pull request by aramesh-msft**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by aramesh-msft** [comment]
Deployment step that failed in pipeline works when I ran it locally, not sure why the pipeline complains about a valid API version 2019-04-01

```
New-AzResourceGroupDeployment -ResourceGroupName aramesh-test-rg -TemplateFile D:\azure-quickstart-templates\101-front-door-create-redirect\azuredeploy.json -verbose

cmdlet New-AzResourceGroupDeployment at command pipeline position 1
Supply values for the following parameters:
(Type !? for Help.)
frontDoorName: testRedirect
backendAddress: www.msn.com
VERBOSE: Performing the operation "Creating Deployment" on target "aramesh-test-rg".
VERBOSE: 11:34:30 PM - Template is valid.
VERBOSE: 11:34:32 PM - Create template deployment 'azuredeploy'
VERBOSE: 11:34:32 PM - Checking deployment status in 5 seconds
VERBOSE: 11:34:38 PM - Checking deployment status in 5 seconds
VERBOSE: 11:34:43 PM - Checking deployment status in 5 seconds
VERBOSE: 11:34:48 PM - Checking deployment status in 5 seconds
VERBOSE: 11:34:54 PM - Checking deployment status in 5 seconds
VERBOSE: 11:34:59 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:04 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:10 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:15 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:20 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:26 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:31 PM - Checking deployment status in 11 seconds
VERBOSE: 11:35:42 PM - Resource Microsoft.Network/frontDoors 'testRedirect' provisioning status is running
VERBOSE: 11:35:42 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:47 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:53 PM - Checking deployment status in 5 seconds
VERBOSE: 11:35:58 PM - Checking deployment status in 5 seconds
VERBOSE: 11:36:03 PM - Checking deployment status in 5 seconds
VERBOSE: 11:36:09 PM - Resource Microsoft.Network/frontDoors 'testRedirect' provisioning status is succeeded


DeploymentName          : azuredeploy
ResourceGroupName       : aramesh-test-rg
ProvisioningState       : Succeeded
Timestamp               : 3/5/2020 7:36:05 AM
Mode                    : Incremental
TemplateLink            :
Parameters              :
                          Name              Type                       Value
                          ================  =========================  ==========
                          frontDoorName     String                     testRedirect
                          backendAddress    String                     www.msn.com

Outputs                 :
DeploymentDebugLogLevel :

```

**Comment by bmoore-msft** [comment]
/frontdoor is not available in US Gov cloud... so add the environments property to metadata.json (see contribution guide)

File: 101-front-door-create-redirect/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        },
+        "backendAddress": {
+            "type": "string",
+            "metadata": {
+                "description": "The hostname of the backend. Must be an IP address or FQDN."
+            }
+        }
+    },
+    "variables": {
+        "frontdoorLocation": "global"
```
Comment (bmoore-msft) [inline 101-front-door-create-redirect/azuredeploy.json]: remove this and just hardcode the location on the resource...

File: 101-front-door-create-redirect/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                                "id": "[resourceId('Microsoft.Network/frontDoors/healthProbeSettings', parameters('frontDoorName'), 'healthProbeSettings1')]"
+                            }
+                        }
+                    }
+                ],
+                "frontendEndpoints": [
+                    {
+                        "name": "frontendEndpoint1",
+                        "properties": {
+                            "hostName": "[concat(parameters('frontDoorName'), '.azurefd.net')]",
```
Comment (bmoore-msft) [inline 101-front-door-create-redirect/azuredeploy.json]: remove the hardcoded '.azurefd.net' (assuming the RP has fixed the self reference)

Comment (aramesh-msft) [reply 101-front-door-create-redirect/azuredeploy.json]: We may have to live with this as RP gives us an error -> HostName is invalid. It must be a valid domain name, IP version 4, or IP version 6.

---
In reply to: [388473695](https://github.com/Azure/azure-quickstart-templates/pull/7271#discussion_r388473695) [](ancestors = 388473695)

**Pull request merged**